WordPress/Classes/Models/BasePost.m

```diff
@@ -181,7 +182,7 @@ - (void)setDateCreated:(NSDate *)localDate
      If the date is now or in the past, and the status is scheduled, set the status
      to published.
      */
-    if (self.date_created_gmt == nil) {
+    if (self.date_created_gmt == nil || [self.date_created_gmt isEqualToDate:self.dateModified]) {
```

Comment:
> PS: by the description of the PR it sounds to me like we should be checking if the post is a draft here too?

I think the extra check for a draft status isn't necessary. From what I can tell only drafts, or perhaps  can have identical values for date created and date modified. They should be different for posts with a scheduled, published, pending, private, trashed, or future status.   Tho now that I think about it I don't believe I tested the trashed/restored use case. I'll add that to the list.